### ISSUE TITLE
Tensorflow Predictor, issue with loading the Tensorflow model

### ISSUE THREAD

**Issue by dimasheva1** [cortexlabs/cortex#1973]
### Version

Cortex 0.30

### Description

When I tried deploy the Tensorflow model with Tensorflow predictor, I had problem with loading model. I think that batch size -1 don`t parse by TF. 

### Configuration

- name: titanic-tensorflow
  kind: RealtimeAPI
  predictor:
    type: tensorflow
    path: predictor.py
    dependencies:
      pip: requirements.txt
    models:
      path: "s3://dmytro.shevchenko/keras_model/"
    

### Steps to reproduce

1. deploy RealtimeAPI 
2. cortex logs titanic-tensorflow

### Expected behavior

Model loaded

### Actual behavior

Model didn`t load

### Stack traces

(error output from `cortex logs <api name>`)

Config entry
	index : 0
	path : /mnt/model/_cortex_default
	name : _cortex_default
	platform : tensorflow
2021-03-17 09:45:36.779987: I tensorflow_serving/model_servers/server_core.cc:464] Adding/updating models.
2021-03-17 09:45:36.779995: I tensorflow_serving/model_servers/server_core.cc:575]  (Re-)adding model: _cortex_default
2021-03-17 09:45:37.509643: I tensorflow_serving/util/retrier.cc:46] Retrying of Reserving resources for servable: {name: _cortex_default version: 1} exhausted max_num_retries: 0
2021-03-17 09:45:37.509715: I tensorflow_serving/core/basic_manager.cc:739] Successfully reserved resources to load servable {name: _cortex_default version: 1}
2021-03-17 09:45:37.509724: I tensorflow_serving/core/loader_harness.cc:66] Approving load for servable version {name: _cortex_default version: 1}
2021-03-17 09:45:37.509734: I tensorflow_serving/core/loader_harness.cc:74] Loading servable version {name: _cortex_default version: 1}
2021-03-17 09:45:37.509760: I external/org_tensorflow/tensorflow/cc/saved_model/reader.cc:31] Reading SavedModel from: /mnt/model/_cortex_default/1
2021-03-17 09:45:37.511243: I external/org_tensorflow/tensorflow/cc/saved_model/reader.cc:54] Reading meta graph with tags { serve }
2021-03-17 09:45:37.511269: I external/org_tensorflow/tensorflow/cc/saved_model/loader.cc:234] Reading SavedModel debug info (if present) from: /mnt/model/_cortex_default/1
2021-03-17 09:45:37.514800: I external/org_tensorflow/tensorflow/cc/saved_model/loader.cc:199] Restoring SavedModel bundle.
2021-03-17 09:45:37.534548: I external/org_tensorflow/tensorflow/cc/saved_model/loader.cc:183] Running initialization op on SavedModel bundle at path: /mnt/model/_cortex_default/1
2021-03-17 09:45:37.540331: I external/org_tensorflow/tensorflow/cc/saved_model/loader.cc:303] SavedModel load for tags { serve }; Status: success: OK. Took 30573 microseconds.
signature defs found in model '_cortex_default' for version '1': {'serving_default': {'inputs': {'input_1': {'name': 'serving_default_input_1:0', 'dtype': 'DT_FLOAT', 'tensorShape': {'dim': [{'size': '-1'}, {'size': '12'}]}}}, 'outputs': {'dense_1': {'name': 'StatefulPartitionedCall:0', 'dtype': 'DT_FLOAT', 'tensorShape': {'dim': [{'size': '-1'}, {'size': '1'}]}}}, 'methodName': 'tensorflow/serving/predict'}, '__saved_model_init_op': {'outputs': {'__saved_model_init_op': {'name': 'NoOp', 'tensorShape': {'unknownRank': True}}}}}
model '_cortex_default' of version '1' couldn't be loaded: error: signature_key was not configured by user, please specify one the following keys 'serving_default, __saved_model_init_op' for model '_cortex_default' of version '1' (found in the signature def map)
2021-03-17 09:45:37.541002: I tensorflow_serving/servables/tensorflow/saved_model_warmup_util.cc:59] No warmup data file found at /mnt/model/_cortex_default/1/assets.extra/tf_serving_warmup_requests
2021-03-17 09:45:37.541904: I tensorflow_serving/util/retrier.cc:46] Retrying of Loading servable: {name: _cortex_default version: 1} exhausted max_num_retries: 0
2021-03-17 09:45:37.542242: I tensorflow_serving/core/loader_harness.cc:87] Successfully loaded servable version {name: _cortex_default version: 1}
2021-03-17 09:45:37.544376: I tensorflow_serving/model_servers/server_core.cc:464] Adding/updating models.
2021-03-17 09:45:37.608717: I tensorflow_serving/core/loader_harness.cc:138] Quiescing servable version {name: _cortex_default version: 1}
2021-03-17 09:45:37.608770: I tensorflow_serving/core/loader_harness.cc:145] Done quiescing servable version {name: _cortex_default version: 1}
2021-03-17 09:45:37.608782: I tensorflow_serving/core/loader_harness.cc:120] Unloading servable version {name: _cortex_default version: 1}
2021-03-17 09:45:37.611080: I ./tensorflow_serving/core/simple_loader.h:363] Calling MallocExtension_ReleaseToSystem() after servable unload with 88135
2021-03-17 09:45:37.611106: I tensorflow_serving/core/loader_harness.cc:128] Done unloading servable version {name: _cortex_default version: 1}

### Additional context

(optional)

### Suggested solution

(optional)

**Comment by RobertLucian** [comment]
@dimasheva1 the error message says `model '_cortex_default' of version '1' couldn't be loaded: error: signature_key was not configured by user, please specify one the following keys 'serving_default, __saved_model_init_op' for model '_cortex_default' of version '1' (found in the signature def map)`.

What this means is that you have to specify the signature key for your model and that can either be `serving_default` or `__saved_model_init_op`. You can set this field in your API spec - the field is called `predictor.models.signature_key` (at the same level as the `path` field for your model).

Also, you can check out the docs on this here as well: https://docs.cortex.dev/workloads/realtime-apis/configuration#tensorflow-predictor

Let us know if this was your problem!

**Comment by vishalbollu** [comment]
Closing due to inactivity. Feel free to reopen this issue or a file a new one.
